frame/contracts/src/tests.rs

```diff
@@ -1762,82 +1783,23 @@ fn lazy_removal_partial_remove_works() {
 }
 
 #[test]
-fn lazy_removal_does_no_run_on_full_block() {
-	let (code, hash) = compile_module::<Test>("self_destruct").unwrap();
+fn lazy_removal_does_no_run_on_full_queue_and_full_block() {
 	ExtBuilder::default().existential_deposit(50).build().execute_with(|| {
-		let min_balance = <Test as Config>::Currency::minimum_balance();
-		let _ = Balances::deposit_creating(&ALICE, 1000 * min_balance);
-
-		assert_ok!(Contracts::instantiate_with_code(
-			Origin::signed(ALICE),
-			min_balance * 100,
-			GAS_LIMIT,
-			None,
-			code,
-			vec![],
-			vec![],
-		),);
-
-		let addr = Contracts::contract_address(&ALICE, &hash, &[]);
-		let info = <ContractInfoOf<Test>>::get(&addr).unwrap();
-		let max_keys = 30;
-
-		// Create some storage items for the contract.
-		let vals: Vec<_> = (0..max_keys)
-			.map(|i| (blake2_256(&i.encode()), (i as u32), (i as u32).encode()))
-			.collect();
-
-		// Put value into the contracts child trie
-		for val in &vals {
-			Storage::<Test>::write(&info.trie_id, &val.0, Some(val.2.clone()), None, false)
-				.unwrap();
-		}
-		<ContractInfoOf<Test>>::insert(&addr, info.clone());
-
-		// Terminate the contract
-		assert_ok!(Contracts::call(
-			Origin::signed(ALICE),
-			addr.clone(),
-			0,
-			GAS_LIMIT,
-			None,
-			vec![]
-		));
-
-		// Contract info should be gone
-		assert!(!<ContractInfoOf::<Test>>::contains_key(&addr));
-
-		let trie = info.child_trie_info();
-
-		// But value should be still there as the lazy removal did not run, yet.
-		for val in &vals {
-			assert_eq!(child::get::<u32>(&trie, &blake2_256(&val.0)), Some(val.1));
-		}
-
 		// Fill up the block which should prevent the lazy storage removal from running.
 		System::register_extra_weight_unchecked(
 			<Test as system::Config>::BlockWeights::get().max_block,
 			DispatchClass::Mandatory,
 		);
 
+		// Fill the deletion queue with dummy values, so that on_initialize attempts
+		// to clear the queue
+		Storage::<Test>::fill_queue_with_dummies();
+
 		// Run the lazy removal without any limit so that all keys would be removed if there
 		// had been some weight left in the block.
-		let weight_used = Contracts::on_initialize(Weight::max_value());
-		let base = <<Test as Config>::WeightInfo as WeightInfo>::on_initialize();
+		let weight_used = Contracts::on_initialize(System::block_number());
+		let base = <<Test as Config>::WeightInfo as WeightInfo>::on_process_deletion_queue_batch();
 		assert_eq!(weight_used, base);
```

Comment:
You can add a check here that the queue is still full.

Reply:
Yes, that makes sense! I added the check! However, I realized that the type `DeletionQueueDepth` is defined as `u32` in the `Config` trait implementation for `Test`. Wouldn't the native type be `usize`?

Reply:
We try to avoid `usize` where possible because it is different in native vs wasm. This is why we rather convert from `usize` to `u32` instead of defing it is `usize`.